### PR TITLE
hw4: advanced components

### PR DESCRIPTION
**Pull request by kamiour**
- Implemented Modal component logic
- Implemented Delete Movie modal
- Implemented Edit Movie reusable modal
- Implemented FormSelect component
- sorting select was implemented in the previous homework
- Used stateless/stateful approach
- Used event handlers
- Used useEffect hook

### CODE REVIEW

File: src/components/DeleteMovieConfirm/DeleteMovieConfirm.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+interface DeleteMovieConfirmProps {
+  handleConfirm: () => void;
+}
+
+const DeleteMovieConfirm = ({ handleConfirm }: DeleteMovieConfirmProps) => {
+  return (
+    <div className="confirm-modal">
+      <p className="confirm-text">Are you sure you want to delete this movie?</p>
+
+      <button className="app-btn confirm-btn" onClick={() => handleConfirm()}>
```
Comment (Twixs) [inline src/components/DeleteMovieConfirm/DeleteMovieConfirm.tsx:12]: use shorthand to pass function

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  useEffect(() => {
+    modalRoot!.appendChild(el);
+    document.body.classList.add('no-scroll');
+
+    return () => {
+      modalRoot!.removeChild(el);
+      document.body.classList.remove('no-scroll');
+    };
+  });
```
Comment (Twixs) [inline src/components/Modal/Modal.tsx:25]: I don't understand why you need additional container. Can't ypu just do `ReactDOM.createPortal(modalLayout, modalRoot)`?

File: src/components/DeleteMovieConfirm/DeleteMovieConfirm.tsx (hunk 1 above)
Comment (kamiour) [reply src/components/DeleteMovieConfirm/DeleteMovieConfirm.tsx]: @Twixs Thank you fixed. Sorry, I thought I fixed it everywhere before...

File: src/components/Modal/Modal.tsx (hunk 2 above)
Comment (kamiour) [reply src/components/Modal/Modal.tsx]: Yep, I can, thank you for pointing this out. I thought I needed that extra container for clearing `modalRoot`. But it gets cleared anyway.

**Pull request merged**